Signal/src/ViewControllers/AddToBlockListViewController.m

```diff
@@ -0,0 +1,344 @@
+                               @"BLOCK_LIST_VIEW_BLOCK_BUTTON", @"A label for the block button in the block list view")
+                  forState:UIControlStateNormal];
+    [_blockButton addTarget:self action:@selector(blockButtonPressed:) forControlEvents:UIControlEventTouchUpInside];
+    [blockButtonRow addSubview:_blockButton];
+    [_blockButton autoCenterInSuperview];
+    [_blockButton autoSetDimension:ALDimensionWidth toSize:160];
+    [_blockButton autoSetDimension:ALDimensionHeight toSize:40];
+
+    [self updateBlockButtonEnabling];
+}
+
+- (UIView *)createRowWithHeight:(CGFloat)height previousRow:(nullable UIView *)previousRow
+{
+    UIView *row = [UIView new];
+    [self.view addSubview:row];
+    [row autoPinWidthToSuperview];
+    if (previousRow) {
+        [row autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:previousRow withOffset:0];
+    } else {
+        [row autoPinToTopLayoutGuideOfViewController:self withInset:0];
+    }
+    [row autoSetDimension:ALDimensionHeight toSize:height];
+    return row;
+}
+
+#pragma mark - Country
+
+- (void)populateDefaultCountryNameAndCode {
+    NSLocale *locale      = NSLocale.currentLocale;
+    NSString *countryCode = [locale objectForKey:NSLocaleCountryCode];
+    NSNumber *callingCode = [[PhoneNumberUtil sharedUtil].nbPhoneNumberUtil getCountryCodeForRegion:countryCode];
+    NSString *countryName = [PhoneNumberUtil countryNameFromCountryCode:countryCode];
+    [self updateCountryWithName:countryName
+                    callingCode:[NSString stringWithFormat:@"%@%@",
+                                 COUNTRY_CODE_PREFIX,
+                                 callingCode]
+                    countryCode:countryCode];
+}
+
+- (void)updateCountryWithName:(NSString *)countryName
+                  callingCode:(NSString *)callingCode
+                  countryCode:(NSString *)countryCode {
+
+    _callingCode = callingCode;
+
+    NSString *title = [NSString stringWithFormat:@"%@ (%@)",
+                       callingCode,
+                       countryCode.uppercaseString];
+    [_countryCodeButton setTitle:title
+                        forState:UIControlStateNormal];
+    [_countryCodeButton layoutSubviews];
+}
+
+- (void)setcallingCode:(NSString *)callingCode
+{
+    _callingCode = callingCode;
+
+    [self updateBlockButtonEnabling];
+}
+
+#pragma mark - Actions
+
+- (void)showCountryCodeView:(id)sender {
+    CountryCodeViewController *countryCodeController = [[UIStoryboard storyboardWithName:@"Registration" bundle:NULL]
+        instantiateViewControllerWithIdentifier:@"CountryCodeViewController"];
+    countryCodeController.delegate = self;
+    countryCodeController.shouldDismissWithoutSegue = YES;
+    UINavigationController *navigationController =
+        [[UINavigationController alloc] initWithRootViewController:countryCodeController];
+    [self presentViewController:navigationController animated:YES completion:[UIUtil modalCompletionBlock]];
```

Comment:
I'd certainly propose that we get rid of all segues at some point (a stepping stone towards removing the storyboards altogether), but I was trying to keep these changes narrow for now since tinkering with segues can have unexpected consequences =(.